src/main/kotlin/com/hibob/academy/resource/PetResource.kt

```diff
@@ -34,7 +34,9 @@ class PetResource {
         @Path("/{petId}/updatePetName")
         fun updateName(@PathParam("petId") petId: Long, @QueryParam("newName") newName: String?): Response {
             val pet = PetData(id = petId, name = "Angie", type = PetType.DOG, companyId = 12L, dateOfArrival = Date(), ownerId = null)
-            pet.name = newName ?: pet.name
+            newName?.let{
+                pet.copy(id = petId, name = newName, type = PetType.DOG, companyId = 12L, dateOfArrival = Date(), ownerId = null)
+            }
```

Comment:
the copy method returns a new instance of the PetData object, but being that you don't store it in a variable just gets lost. instead, i recommend something like this:
```suggestion
            val petName = newName?.let{ it } ?: "Angie"
            val pet = PetData(id = petId, name = petName, type = PetType.DOG, companyId = 12L, dateOfArrival = Date(), ownerId = null)
```

Reply:
Thanks, fixed it